src/api_proxy/service_control/request_builder_test.cc

```diff
@@ -367,17 +422,56 @@ TEST_F(RequestBuilderTest, FillReportRequestEmptyOptionalTest) {
   ASSERT_EQ(expected_text, text);
 }
 
-TEST_F(RequestBuilderTest, CredentailIdApiKeyTest) {
+TEST_F(RequestBuilderTest, CredentailIdApiKeyVerifiedTest) {
   ReportRequestInfo info;
   FillOperationInfo(&info);
 
+  info.api_key_state = api_key::ApiKeyState::VERIFIED;
+
   gasv1::ReportRequest request;
   ASSERT_TRUE(scp_.FillReportRequest(info, &request).ok());
 
+  ASSERT_TRUE(request.operations(0).labels().contains("/credential_id"));
   ASSERT_EQ(request.operations(0).labels().at("/credential_id"),
             "apikey:api_key_x");
 }
 
+TEST_F(RequestBuilderTest, CredentailIdApiKeyInvalidTest) {
+  ReportRequestInfo info;
+  FillOperationInfo(&info);
+
+  info.api_key_state = api_key::ApiKeyState::INVALID;
+
+  gasv1::ReportRequest request;
+  ASSERT_TRUE(scp_.FillReportRequest(info, &request).ok());
+
+  ASSERT_FALSE(request.operations(0).labels().contains("/credential_id"));
```

Comment:
Can we verify all 3 fields:  consumer_id, credentail_id, and log_entry.  

Verify them for all api_key_states?

Reply:
+1

Reply:
Done